backend/src/impl/db_utils/dataset_db_utils.py

```diff
@@ -1,42 +1,125 @@
+                    "dataset_id": dataset_id,
+                    "dataset_name": dataset_name,
+                    "sub_dataset": None if sub_dataset == "__NONE__" else sub_dataset,
+                    "split": v_sub["splits"],
+                    "tasks": tasks,
+                }
+                self.metadatas.append(DatasetMetadata.from_dict(doc))
+        self.name_trie = marisa_trie.Trie(self.name_dict.keys())
+
+
 class DatasetDBUtils:
+
+    online_path = "https://raw.githubusercontent.com/ExpressAI/DataLab/main/utils/dataset_info.jsonl"  # noqa
+    _cached_db: DatasetDB | None = None
+    _cached_time: datetime | None = None
+    _cached_lifetime: timedelta = timedelta(hours=6)
+
+    @staticmethod
+    def get_dataset_db() -> DatasetDB:
+        """
+        Get a collection of datasets, indexed by dataset name, containing every
+        sub_dataset
+        """
+        if DatasetDBUtils._cached_db is None or (
+            datetime.now() - unwrap(DatasetDBUtils._cached_time)
+            > unwrap(DatasetDBUtils._cached_lifetime)
+        ):
+            local_path = cache_online_file(
+                DatasetDBUtils.online_path,
+                "info/dataset_info.jsonl",
+                lifetime=DatasetDBUtils._cached_lifetime,
+            )
+            datasets = {}
+            with open(local_path, "r") as fin:
+                for line in fin:
+                    data = json.loads(line)
+                    for k, v in data.items():
+                        # skip 'ERROR' or 'SKIPPED' entries
+                        if isinstance(v, dict) and len(v) > 0:
+                            datasets[k] = v
+            DatasetDBUtils._cached_db = DatasetDB(datasets)
+            DatasetDBUtils._cached_time = datetime.now()
+        return DatasetDBUtils._cached_db
+
+    @staticmethod
+    def find_dataset_by_id(dataset_id: str) -> DatasetMetadata | None:
+        my_db = DatasetDBUtils.get_dataset_db()
+        _id = my_db.id_dict.get(dataset_id)
+        return my_db.metadatas[_id] if _id is not None else None
+
     @staticmethod
     def find_datasets(
-        page: int,
-        page_size: int,
+        page: int = 0,
+        page_size: int = 0,
         dataset_ids: Optional[list[str]] = None,
         dataset_name: Optional[str] = None,
         task: Optional[str] = None,
         no_limit: bool = False,
     ) -> DatasetsReturn:
-        """
-        fuzzy match works like a `LIKE {name_prefix}%` operation now. can extend this
-        and allow for full text search in the future.
-          - `no_limit=True` ignores page and page_size to retrieve unlimited records.
-            This option should not be exposed to users.
-        """
-        filt: dict[str, Any] = {}
+        my_db = DatasetDBUtils.get_dataset_db()
+        my_ids: set[int] | None = None
         if dataset_ids is not None:
-            filt["_id"] = {"$in": [ObjectId(_id) for _id in dataset_ids]}
+            my_ids = set()
```

Comment:
`_id` is identical to the `i` in `DatasetDB`'s  `__init__` function. We can standardize the variable name. Same for `my_ids`.